core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/registry/CachingCodecRegistry.java

```diff
@@ -279,9 +291,74 @@ protected boolean matches(TypeCodec<?> codec, GenericType<?> javaType, boolean i
     }
   }
 
+  protected GenericType<?> inferJavaTypeFromCqlType(@NonNull DataType cqlType) {
+    if (cqlType instanceof ListType) {
+      DataType elementType = ((ListType) cqlType).getElementType();
+      return GenericType.listOf(inferJavaTypeFromCqlType(elementType));
```

Comment:
don't we want to provide some upper bound for that recursion?
Is it possible that the user will create so many enclosing collections that our code will throw `StackOverflow`?

Reply:
I think it would blow up server-side first, because you would need to create a collection type with thousands of nested types for that to happen, as in `list<list<list<list<...`.